src/internal/m365/backup.go

```diff
@@ -158,13 +158,21 @@ func (ctrl *Controller) IsBackupRunnable(
 	resourceOwner string,
 ) (bool, error) {
 	if service == path.SharePointService {
-		// No "enabled" check required for sharepoint
+		_, err := ctrl.AC.Sites().GetRoot(ctx)
+		if err != nil {
+			if clues.HasLabel(err, graph.LabelsNoSharePointLicense) {
+				return false, clues.Stack(graph.ErrServiceNotEnabled, err)
```

Comment:
I'm kind of worried that we're heading towards a split in how things being disabled is treated between different services. For Exchange we seem to be checking errors directly with functions that return bools. For SharePoint we seem to be adding labels to errors and assuming something upstream from that addition will check labels

Reply:
I'd say it's more like:
* api and graph handlers will return errors annotated with sentinels and labels.
* the next layer up (ie: anything in m365) should catch those sentinels and labels and handle them appropriately, returning an error if needed.
* All layers above that (operations/cli/sdk/etc) should only need to check errors.

Is the choice to use a label over a stacked error a little fuzzy?  Yeah, it is.  I don't think we have a clear pattern for prioritizing one or the other.  On the other hand, I don't think it's causing us much harm, either, since the extent to which a layer needs to know the difference is somewhat well contained.  There's outliers we can probably watch for smells; but I don't think we're into the need for refactoring yet.

Reply:
> On the other hand, I don't think it's causing us much harm, either, since the extent to which a layer needs to know the difference is somewhat well contained

I'm not sure I completely agree with this. I don't work on the m365 package quite as much so I'm honestly not completely sure where one layer ends and another begins. I think bugs like needing to use `errors.As` instead of a direct type cast are symptoms of not having clear interface definitions and standardized patterns. Because bugs like that are dependent on both internal e2e system behavior and inconsistent behavior by external services it's extra difficult to find and catch them. I only managed to catch the one I did because CI happened to fail

When people work in areas they aren't familiar with they're more likely to copy and modify code near what they're doing (or at least code doing similar things) to fit their needs. Standardization makes this simpler because then whether a solution works or not isn't (as) dependent on which source the "example" code happened to come from